### ISSUE TITLE
passive mode without selection pastes clipboard contents.

### ISSUE THREAD

**Issue by NeoTeo** [federico-terzi/espanso#213]
Triggering the passive mode key combo (eg. CTRL twice) without having selected anything will paste the contents of the clipboard at the caret.

**Comment by federico-terzi** [comment]
Hey @NeoTeo,

Thank you very much for the feedback, this is definitely a bug! I'll do my best to fix it in the next release and I'll let you know when it's ready to be tried out.

Cheers :)

**Comment by AndydeCleyre** [comment]
I think a problem with this fix is that if the user copies the selection to clipboard before hitting the passive trigger key (e.g. as a backup of the content before seeing how it's replaced), it will not be replaced.

**Comment by federico-terzi** [comment]
> I think a problem with this fix is that if the user copies the selection to clipboard before hitting the passive trigger key (e.g. as a backup of the content before seeing how it's replaced), it will not be replaced.

This is true indeed, but I couldn't come up with a better solution. Do you have any other idea?

**Comment by AndydeCleyre** [comment]
Would this work?

- save initial clipboard contents (`A`)
- set clipboard to empty string
- trigger keyboard copy to clipboard
- if clipboard is empty, restore `A` to clipboard then abort; else, perform replacement then restore `A` to clipboard

**Comment by federico-terzi** [comment]
@AndydeCleyre yes, your solution could partially solve the problem. The only problem I see is that if the user has currently an image copied in the clipboard, this process won't be able to recover it. Could you please create an issue with your proposed solution so that we can keep track of it?

Thanks for your help :)
